Tests/VaporTests/URITests.swift

```diff
@@ -132,6 +132,19 @@ final class URITests: XCTestCase {
         XCTAssertURIComponents(scheme: "wss", host: "echo.websocket.org", path: "/", generate: "wss://echo.websocket.org/")
     }
     
+    func testSpecialCharacters() {
+        XCTAssertURIString(
+            "https://user:pass@vapor.codes:1234/foo;?bar=abcd%C3%A4%2B#qux",
+            hasScheme: "https",
+            hasUserinfo: "user:pass",
+            hasHost: "vapor.codes",
+            hasPort: 1234,
+            hasPath: "/foo;",
+            hasQuery: "bar=abcd%C3%A4%2B",
+            hasFragment: "qux"
+        )
```

Comment:
```suggestion
    func testSemicolonIsNotEncodedInPathComponent() {
        XCTAssertURIString(
            "https://user:pass@vapor.codes:1234/foo;?bar=abcd%3B%C3%A4%2B#qux%3B",
            hasScheme: "https",
            hasUserinfo: "user:pass",
            hasHost: "vapor.codes",
            hasPort: 1234,
            hasPath: "/foo;",
            hasQuery: "bar=abcd%3B%C3%A4%2B",
            hasFragment: "qux%3B"
        )
```
^ so that we test that we're _specifically_ only mutating the path component.

Reply:
I also slightly adapted the query so that we test that we can have `;` and `%3B` at the same time:

```
hasQuery: "bar=abcd%3B%C3%A4%2B;efg",
```